examples/embeddable_examples/public/react_embeddables/search/search_react_embeddable.tsx

```diff
@@ -54,51 +53,49 @@ export const getSearchEmbeddableFactory = (services: Services) => {
       );
 
-      let isUnmounted = false;
       const error$ = new BehaviorSubject<Error | undefined>(undefined);
       const count$ = new BehaviorSubject<number>(0);
-      const onFetch = (fetchContext: FetchContext, isCanceled: () => boolean) => {
-        error$.next(undefined);
-        if (!defaultDataView) {
-          return;
-        }
-        dataLoading$.next(true);
-        getCount(
-          defaultDataView,
-          services.data,
-          fetchContext.filters ?? [],
-          fetchContext.query,
-          // timeRange and timeslice provided seperatly so consumers can decide
-          // whether to refetch data for just mask current data.
-          // In this example, we must refetch because we need a count within the time range.
-          fetchContext.timeslice
-            ? {
-                from: new Date(fetchContext.timeslice[0]).toISOString(),
-                to: new Date(fetchContext.timeslice[1]).toISOString(),
-                mode: 'absolute' as 'absolute',
-              }
-            : fetchContext.timeRange
-        )
-          .then((nextCount: number) => {
-            if (isUnmounted || isCanceled()) {
+      const fetchSubscription = fetch$(api)
+        .pipe(
+          switchMap(async (fetchContext) => {
+            error$.next(undefined);
+            if (!defaultDataView) {
               return;
             }
-            dataLoading$.next(false);
-            count$.next(nextCount);
-          })
-          .catch((err) => {
-            if (isUnmounted || isCanceled()) {
-              return;
+
+            try {
+              dataLoading$.next(true);
+              const count = await getCount(
+                defaultDataView,
+                services.data,
+                fetchContext.filters ?? [],
+                fetchContext.query,
+                // timeRange and timeslice provided seperatly so consumers can decide
+                // whether to refetch data for just mask current data.
+                // In this example, we must refetch because we need a count within the time range.
+                fetchContext.timeslice
+                  ? {
+                      from: new Date(fetchContext.timeslice[0]).toISOString(),
+                      to: new Date(fetchContext.timeslice[1]).toISOString(),
+                      mode: 'absolute' as 'absolute',
+                    }
+                  : fetchContext.timeRange
+              );
+              return { count };
+            } catch (error) {
+              return { error };
             }
-            dataLoading$.next(false);
-            error$.next(err);
-          });
-      };
-      const unsubscribeFromFetch = onFetchContextChanged({
-        api,
-        onFetch,
-        fetchOnSetup: true,
-      });
+          })
+        )
+        .subscribe((next) => {
+          dataLoading$.next(false);
+          if (next && next.hasOwnProperty('count')) {
+            count$.next(next.count as number);
```

Comment:
nit: Do you need this cast here? Hopefully your switchmap above will resolve to `{ count: number } | {error: Error }`. If not, it should be made to do so with a type assertion.